code-challenges/map/challenges-07.test.js

```diff
@@ -0,0 +1,287 @@
+'use strict';
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 1
+
+Write a function named forLoopTwoToThe that, given an array of integers as input, iterates over the array and returns a new array. The returned array should contain the result of raising 2 to the power of the original input element.
+
+You may choose to complete this challenge using a for loop, for...in syntax, or for...of syntax.
+
+For example, twoToThe([1,2,3]) returns [2,4,8] because 2 ^ 1 = 2, 2 ^ 2 = 4, and 2 ^ 3 = 8.
+------------------------------------------------------------------------------------------------ */
+
+const forLoopTwoToThe = (arr) => {
+  let result = [];
+  for(let i = 0; i < arr.length; i++){
+    result.push(Math.pow(2,arr[i])); 
+  }
+  return result;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 2
+
+Write a function named forEachTwoToThe that produces the same output as your forLoopTwoToThe function from challenge 1, but uses forEach instead of a for loop.
+------------------------------------------------------------------------------------------------ */
+
+const forEachTwoToThe = (arr) => {
+  let result = [];
+  arr.forEach((value)=>{
+    result.push(Math.pow(2,value));
+  });
+  return result;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 3
+
+Write a function named mapTwoToThe that produces the same output as your forLoopTwoToThe function from challenge 1 and your forEachTwoToThe function from challenge 2, but uses map instead of a for loop or forEach.
+------------------------------------------------------------------------------------------------ */
+
+const mapTwoToThe = (arr) => {
+  // Solution code here...
+  let result = [];
+  result = arr.map(x=> Math.pow(2,x) );
+  return result;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 4
+
+Write a function named charCode that, given an array of letters as an input, uses map to return a new array where each element is the result of the `charCodeAt` method on the original array element.
+
+Read the MDN documentation on String.charCodeAt() if necessary.
+
+For example: charCode(['h','i']) returns [104, 105].
+------------------------------------------------------------------------------------------------ */
+
+const charCode = (arr) => {
+  // Solution code here...
+  let result = [];
+  arr.map(function (str){
```

Comment:
Like we discussed earlier in lecture, here you are using .map() like .forEach(). When using .map() you shouldn't need to use .push()